esphome/components/deep_sleep/deep_sleep_component.h

```diff
@@ -108,11 +111,11 @@ class DeepSleepComponent : public Component {
 
   optional<uint64_t> sleep_duration_;
 #ifdef USE_ESP32
-  InternalGPIOPin *wakeup_pin_;
-  WakeupPinMode wakeup_pin_mode_{WAKEUP_PIN_MODE_IGNORE};
+  std::vector<WakeupPinItem> wakeup_pins_;
   optional<Ext1Wakeup> ext1_wakeup_;
   optional<bool> touch_wakeup_;
   optional<WakeupCauseToRunDuration> wakeup_cause_to_run_duration_;
+  bool prepare_pin_(esphome::InternalGPIOPin * /*pin*/, WakeupPinMode /*pin_mode*/);
```

Comment:
To keep it simpler, we just give the args their names inline
```suggestion
  bool prepare_pin_(esphome::InternalGPIOPin * pin, WakeupPinMode pin_mode);
```